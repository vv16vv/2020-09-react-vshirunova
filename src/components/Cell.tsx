import React from "react";
import './../css/styles.css';

interface CellProp {
    x: number;
    y: number;
    filled: boolean;
}

export const Cell: React.FC<CellProp> = (({x, y, filled}) => {
    return (
        <td className={filled ? "filledCell" : "emptyCell"}>
            {`${x}, ${y}`}
        </td>
    )
});