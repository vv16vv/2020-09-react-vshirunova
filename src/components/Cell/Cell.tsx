import React from "react";

import {StyledTd} from "@/components/styled/StyledComponents";

interface CellProp {
    x: number;
    y: number;
    filled: boolean;
    clickHandler: (x: number, y: number) => void;
}

export const Cell: React.FC<CellProp> = (({x, y, filled, clickHandler}) => {
    const filledSign = filled ? "X" : " ";
    const id = `id${x}-${y}`;
    return (
        <StyledTd filled={filled} id={id} onClick={() => clickHandler(x, y)}>
            {filledSign}
        </StyledTd>
    )
});