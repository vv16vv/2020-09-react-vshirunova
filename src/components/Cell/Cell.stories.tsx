import React from "react";
import { action } from "@storybook/addon-actions";
import {boolean, number, withKnobs} from "@storybook/addon-knobs";

import {Cell} from "./Cell";

export default {
    title: "Cell Story",
    decorators: [withKnobs],
}

const GeneralCellStory: React.FC<{x: number, y: number, filled: boolean}> = (props) => {
    const x = number('X', props.x);
    const y = number('Y', props.y);
    const filled = boolean('Is Filled', props.filled);
    return (
        <Cell x={x} y={y} filled={filled} clickHandler={action("Cell clicked")}/>
    );
};

export const FilledCellStory: React.FC<{}> = () => {
    return <GeneralCellStory x={3} y={4} filled={true}/>
};

export const EmptyCellStory: React.FC<{}> = () => {
    return <GeneralCellStory x={3} y={4} filled={false}/>
};
