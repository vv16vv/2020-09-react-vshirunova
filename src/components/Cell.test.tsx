import React from "react";
import {render} from "enzyme";
import {Cell} from "./Cell";

describe("Cell", () => {
    it("coordinates are shown, cell is filled up", () => {
       expect(render(<Cell x={3} y={4} filled={true} />).text()).toEqual("[X] 3, 4")
    });

    it("coordinates are shown, cell is empty", () => {
       expect(render(<Cell x={3} y={4} filled={false} />).text()).toEqual("[ ] 3, 4")
    });
});