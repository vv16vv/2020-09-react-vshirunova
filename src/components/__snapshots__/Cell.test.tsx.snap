// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should show coordinates and empty sign 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-3chxdl{text-align:center;width:1.5em;height:1.5em;border-width:1px;border-style:solid;border-color:#000000;background-color:#ffffff;color:#00008b;}",
      }
    }
    data-emotion="css 3chxdl"
  />,
  <td
    className="css-3chxdl"
    id="id3-4"
    onClick={[Function]}
  >
     
  </td>,
]
`;

exports[`Cell should show coordinates and filled up sign 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-14h0qql{text-align:center;width:1.5em;height:1.5em;border-width:1px;border-style:solid;border-color:#000000;background-color:#00008b;color:#ffffff;}",
      }
    }
    data-emotion="css 14h0qql"
  />,
  <td
    className="css-14h0qql"
    id="id3-4"
    onClick={[Function]}
  >
    X
  </td>,
]
`;
